river/metrics/q0.py

```diff
@@ -0,0 +1,177 @@
+import math
+
+from scipy.special import factorial
+
+from . import base
+
+__all__ = ["Q0", "Q2"]
+
+
+class Q0(base.MultiClassMetric):
+    r"""Q0 index.
+
+    Dom's Q0 measure [^2] uses conditional entropy to calculate the goodness of
+    a clustering solution. However, this term only evaluates the homogeneity of
+    a solution. To measure the completeness of the hypothesized clustering, Dom
+    includes a model cost term calculated using a coding theory argument. The
+    overall clustering quality measure presented is the sum of the costs of
+    representing the data's conditional entropy and the model.
+
+    The motivation for this approach is an appeal to parsimony: Given identical
+    conditional entropies, H(C|K), the clustering solution with the fewest clusters
+    should be preferred.
+
+    The Q0 measure can be calculated using the following formula [^1]
+
+    $$
+    Q_0(C, K) = H(C|K) + \frac{1}{n} \sum_{k=1}^{|K|} \log \binom{h(c) + |C| - 1}{|C| - 1}.
+    $$
+
```

Comment:
Should we consider a note regarding the complexity of this metric?

Reply:
I have put a note here, and in the normalized version of the metric (Q2), saying that this is one order of magnitude slower than most other implemented metrics.